orbit/cmd/orbit/orbit.go

```diff
@@ -645,6 +689,178 @@ func (d *desktopRunner) interrupt(err error) {
 	}
 }
 
+// shell out to osquery, and get the system uuid
+func getUUID(osqueryPath string) (string, error) {
+	type UuidOutput struct {
+		UuidString string `json:"uuid"`
+	}
+
+	args := []string{"-S", "--json", "select uuid from system_info"}
+	out, err := exec.Command(osqueryPath, args...).Output()
+	if err != nil {
+		return "", err
+	}
+	var uuids []UuidOutput
+	err = json.Unmarshal(out, &uuids)
+	if err != nil {
+		return "", err
+	}
+
+	return uuids[0].UuidString, nil
+}
+
+// getOrbitNodeKeyOrEnroll attempts to read the orbit node key if the file exists on disk
+// otherwise it enrolls the host with Fleet and saves the node key to disk
+func getOrbitNodeKeyOrEnroll(orbitClient *service.Client, rootDir string, enrollSecret string, uuidStr string) (string, error) {
+	nodeKeyFilePath := filepath.Join(rootDir, "secret-node-key.txt")
+	orbitNodeKey, err := readOrbitNodeKey(nodeKeyFilePath)
+	if err == nil {
+		return orbitNodeKey, nil
+	}
```

Comment:
We should check the errors, something like:
```go
switch {
case err == nil:
  return orbitNodeKey, nil
case errors.Is(err, fs.ErrNotExist):
  // OK
default:
  return "", fmt.Errorf("read orbit node key: %w", err)
}
```
And let's just call `os.ReadFile` here directly (no point in wrapping it in `readOrbitNodeKey`).
```suggestion
	}
```
```suggestion
	}
```
```suggestion
	}
```